src/standard_library/array_methods.md

```diff
@@ -0,0 +1,139 @@
+
+# Array
+
+For convenience, the STD provides some ready-to-use, common methods for arrays[^migrationNote]:
+
+## len
+
+Returns the length of an array
+
+```rust
+fn len(_array: Self) -> comptime Field
+```
+
+example
+
+```rust
+// arr = [42, 42]
+fn main(arr : pub [Field; 2]) {
+    let t = arr.len();
+    constrain t == 2;
+}
+```
+
+## sort
+
+Sorts the array with a built-in function
+
+```rust
+fn sort(_array: Self) -> Self
+```
+
+example
+
+```rust
+// arr = [42, 32]
+fn main(arr : pub [u32; 2]) {
+    let t = arr.sort();
+    constrain t == [32, 42];
+}
+```
+
+## sort_via
+
+Sorts the array with a custom sorting function
+
+```rust
+fn sort_via(mut a: Self, ordering: fn(T, T) -> bool) -> Self
+```
+
+example
+
+```rust
+// arr = [42, 32]
+fn main(arr : pub [u32; 2]) {
+    let t = arr.sort_via(|a, b| a < b);
+    constrain t == [32, 42]; // verifies
+
+    let t = arr.sort_via(|a, b| a > b);
+    constrain t == [32, 42]; // does not verify
+
+}
+```
+
+## fold
+
+Applies a function to each element of the array, returning the final accumulated value. First parameter is the initial value
+
+```rust
+fn fold<U>(self, mut accumulator: U, f: fn(U, T) -> U) -> U
+```
+
+example:
+
+``` rust
+
+// arr = [2,2,2,2,2]
+fn main(arr : [Field; 5]) {
```

Comment:
I do think we should prefer to have the literals in the code in general. Having them as parameters is probably closer to what users would actually write but I think including the literals directly instead of going through parameters would make the examples clearer.